.github/workflows/deploy_api.yml

```diff
@@ -0,0 +1,32 @@
+name: Deploy API to Azure App Service
+
+on:
+  workflow_dispatch:
```

Comment:
VNET integration is added as part of terraform creation scripts; this will tie into that. 
It's manual triggering for now, since we didn't have a dedicated subscription for this. There is another story that ties this and TRE terraform creation into a full release into dev subscription (5410)